apps/minifront/src/state/status.ts

```diff
@@ -10,9 +10,15 @@ export interface StatusSlice {
 
 export const createStatusSlice = (): SliceCreator<StatusSlice> => set => ({
   start: async () => {
-    const stream = viewClient.statusStream({});
-
     try {
+      //set the current status using unary request, since statusStream does not guarantee that we will get the status
+      const status = await viewClient.status({});
+      set(state => {
+        state.status.fullSyncHeight = status.fullSyncHeight;
+        if (!status.catchingUp) state.status.latestKnownBlockHeight = status.fullSyncHeight;
+      });
+
+      const stream = viewClient.statusStream({});
```

Comment:
The point is that `statusStream` sends new data to stream only when a new block is detected. 
This means that under normal sync waiting for data from `statusStream` can take up to 5 seconds (time of new block generation). 
And if the network is halted, for example, you will never get data from `statusStream`

Reply:
Ah, that's a good catch! Think this github comment is clearer than the code comment though. 

```diff
- //set the current status using unary request, since statusStream does not guarantee that we will get the status
+ // statusStream sends new data to stream only when a new block is detected. This can take up to 5 seconds (time of new block generated). Therefore, we need to do a unary request to start us off.
```